### ISSUE TITLE
Converter seems to be working incorrectly

### ISSUE THREAD

**Issue by Aidoneus** [kvark/vange-rs#122]
[Test case 1](https://drive.google.com/file/d/1nkqpTv9W5qGeHHqzMAend1AZsjsZaAFt/view?usp=sharing)
Encountered on a different custom world, but here I've managed to reproduce the bug with simple height/material maps. I've included in the archive both the source maps and the resulting output.vmp. In the game I'm unable to pass under the "bridge", as if it was not there – and indeed there is none, when I am converting the resulting output.vmp back into images. I see an area of stripes on the lower level. Upper one somehow disappeared upon convertation.

[Test case 2](https://drive.google.com/file/d/1bMOvO1vGEZnXAKSZS-lLI1qbbj6F73jo/view?usp=sharing)
Another archive with source images and resulting output.vmp. The problem here is that even though I am using two completely identical material maps, which are filled with single color, resulting world somehow have more materials, which can easily be seen if you convert the output.vmp back. As far as I've noticed – this only happens in places, where there are two layers.

So maybe these two bugs are somehow connected?

**Comment by kvark** [comment]
Thank you for filing this with great detail!
I believe this is fixed with 1293df69f737f3e9e7f41b0596234c52874dbcfd
<img width="1013" alt="Screen Shot 2020-08-18 at 00 27 47" src="https://user-images.githubusercontent.com/107301/90470559-d4c67880-e0e9-11ea-83da-b72a9fe70b2d.png">
Please re-open without hesitation if you see more issues!

**Comment by Aidoneus** [comment]
Yeah, the "bridge" from the first example is functional now, thanks!

But now the issue described in the second case arises here. Bridge's upper level now have material 0 after conversion, for some reason. And there are a few areas of material 3, even though the source material images were filled with single color – that of material 1. [Here](https://drive.google.com/file/d/1lmAV_4EiZ33fR2h2Ryo2BFPJ_n5WMubx/view?usp=sharing) I've uploaded an archive with the same source materials and what I've got after converting it back – which is what I see in the original game as well.
